index.js

```diff
@@ -1,72 +1,154 @@
 
-function getYears(/* code here */) {
-
-    /* code here */
+function getYears(callback) {
+    let years = callback.map(callback =>
+        callback.Year)
 
+    return years;
 };
 
-getYears();
-
-/* Task 4: Implement a higher-order function called `getWinners`, that accepts the callback function `getFinals()` and determine the winner (home or away) of each `finals` game. Return the name of all winning countries in an array called `winners` */ 
-
-function getWinners(/* code here */) {
-
-    /* code here */
-
+console.log(getYears(getFinals(fifaData)));
+
+/* Task 4: Implement a higher-order function called `getWinners`, that accepts the callback function `getFinals()` and determine the winner (home or away) of each `finals` game. Return the name of all winning countries in an array called `winners` */
+
+function getWinners(func) {
+    let winners = [];
+    func.forEach( element => {
+        if(element['Home Team Goals'] > element['Away Team Goals']){
+            winners.push(element['Home Team Name']);
+        }
+        else if(element['Away Team Goals'] > element['Home Team Goals']){
+            winners.push(element['Away Team Name']);
+        }
+        else{
+            winners.push(element['Win conditions'].split(' ')[0]);
+        }
+    })
+    return winners;
 };
 
-getWinners();
+console.log(getWinners(getFinals(fifaData)));
 
-/* Task 5: Implement a higher-order function called `getWinnersByYear` that accepts the following parameters and returns a set of strings "In {year}, {country} won the world cup!" 
 
-Parameters: 
+/* Task 5: Implement a higher-order function called `getWinnersByYear` that accepts the following parameters and returns a set of strings "In {year}, {country} won the world cup!"
+
+Parameters:
  * callback function getWinners
  * callback function getYears
  */
 
-function getWinnersByYear(/* code here */) {
-
+function getWinnersByYear(func1, func2) {
+    let newArr = [];
+    for(let i = 0; i < func1.length; i++){
+        newArr.push(`In ${func2[i]}, ${func1[i]} won the world cup!`);
+    }
+    return newArr;
 };
 
-getWinnersByYear();
+console.log(getWinnersByYear(getWinners(getFinals(fifaData)), getYears(getFinals(fifaData))));
 
 /* Task 6: Write a function called `getAverageGoals` that accepts a parameter `data` and returns the the average number of home team goals and away team goals scored per match (Hint: use .reduce and do this in 2 steps) */
 
-function getAverageGoals(/* code here */) {
-
-    /* code here */
-
+function getAverageGoals(data) {
+    let home = [];
+    let away = [];
+    data.forEach(element =>{
+        home.push(element['Home Team Goals']);
+        away.push(element['Away Team Goals']);
+    })
+    const homeTotal = home.reduce( (a, b) => a + b);
+    const awayTotal = away.reduce( (a, b) => a + b);
+
+    return {'Average Goals - Home Team': homeTotal/home.length, 'Away Team': awayTotal/away.length }
```

Comment:
This task has been executed correctly, but for future reference, we want to avoid the long floating-point numbers by using `Math.round()` until we actually need those floating points. This is just a tip by the way, I wasn't expecting you to know to use this 🙂 

Below would return: `Home Team: 2` vs `Home Team: 1.8119858989424207`
```
  return {
    'Average Goals - Home Team': Math.round(homeTotal / home.length),
    'Away Team': Math.round(awayTotal / away.length),
  }
```